### ISSUE TITLE
Todo: Remove aliases that are "also works with" not true aliases

### ISSUE THREAD

**Issue by joshgoebel** [highlightjs/highlight.js#3049]
Ref: https://github.com/highlightjs/highlight.js/pull/3035

This includes things like `zsh` for bash (we list no other shells) and aliasing all the many Lisp/Scheme implementations out there back to our Lisp/Scheme grammar. This is a something for implementors to do if they would prefer these aliases.

Remove:

- [x] zsh
- [x] mysql, oracle
- [x] 'freepascal',
- [x] 'lazarus',
- [x] 'lpr', (lazarus extension)
- [x] 'lfm' (lazarus extension)

**Comment by joshgoebel** [comment]
Copying over the key part of my response from the other thread.

---

Are you aware of us doing something similar with any of our other aliases (perhaps we do)?  If so then perhaps it's time we reviewed the entire policy around this (with v11 around the corner) vs dealing with this as a one-off.

I don't think we do this as a rule, the closest I found:

- zsh for bash
- mysql and oracle for sql_more (but this whole grammar is known to be problematic and it's on the way out)

I think bash is the closest comparison, and I think `zsh` as an alias is wrong there also - and should be removed.  We specifically detect 10 or so different shells inside bash, yet we don't have 10 aliases - nor should we.  Someone can specify `sh` to indicate shell script, or manually register aliases themselves.  And while we do kind of acknowledge that `bash` is perhaps a bit general - we're also not beholden to support every shell's idiosyncrasies - leaving it `bash` helps make that clear.

In general I feel this is an anti-pattern and recommend we remove `zsh` from bash with version 11.

---

Oops, missed Delpi... I think we should consider removing:

- 'freepascal',
- 'lazarus',
- 'lpr', (lazarus extension)
- 'lfm' (lazarus extension)

Leaving only Delphi (the actual grammar) and Pascal variants.  Pascal because it's the "original" language... Lazarus and FreePascal and simply variants of Pascal and shouldn't be directly aliased by Delphi.

XML is a little crazy also, but I also feel there is no obligation created there since all the listed formats are 100% XML by the very nature of XML.  So they bother me less and I'm not sure they are in the same category.

```
    aliases: [
      'html',
      'xhtml',
      'rss',
      'atom',
      'xjb',
      'xsd',
      'xsl',
      'plist',
      'wsf',
      'svg'
    ],
```
